File: AI/code_01/library/specific/Python/FileNames_v01.py
import numpy,os,pandas,re
def FileNames(mfilename,FilterName):
    ## Start
    # Splitting current file name
    [mfilenameAlg,mfilenameVer]=mfilename.split("_")
    [mfilenameVer,_]=mfilenameVer.split(".")
    ## End
    ## Start (Manual entry required)
    # Assigning ML algorithm name based on the file name.    
    if  bool(re.match(re.compile("AI"), mfilename)):    
        AlgName='LSVM' # Algorithm name
    elif bool(re.match(re.compile("BI"), mfilename)):          
        AlgName='GSVM' 
    elif bool(re.match(re.compile("CI"), mfilename)):          
        AlgName='MLP'         
    elif bool(re.match(re.compile("DI"), mfilename)):          
        AlgName='RF'          
    else:
        AlgName='Unknown Alg.' 
    ## End

    ## Start
    # Location of current file
    LocalFolder = os.getcwd()
    LocalFolderFilesTemp = LocalFolder+'\\Files'+'\\Temp'
    ## End
       
    ## Start (Manual entry required)
    # Beamforming selection algorithm     
    BSelecAlg='DI' # Beamforming selection algorithm used for AI training&test DI: Linear_Joint_Init_Iter_Rate_BCC
    BSelecAlgVer='v01' # Beamforming selection algorithm version        
    InputFolderx='B:\Matlab\Shared\BeamSelection\code_01\Results' # Beamforming selection algorithm's results
    ## End

    ## Start
    # Network architectures to be tested
    InputFoldery='B:\\Matlab\\Shared\\Generator\\'
    fnamex = [x for x in os.listdir(InputFoldery) if x.endswith(".xlsx")]
    fname = fnamex[0]
    fname=InputFoldery+fname
    datar = pandas.read_excel(fname, sheet_name='Sheet1')    
    datarx = pandas.read_excel(fname, sheet_name='AlgorithmsList_v01', usecols="C,D", header=None)
    counter=0
    for x in datarx[2]:
        res= BSelecAlg+'_'+BSelecAlgVer==x
        if res:
            BSelecAlgName=datarx.iloc[counter,1]
        counter=counter+1
    BSelecAlgName=BSelecAlgName+'_'+FilterName
    selected_boolean=datar['Runners']==mfilename
    selected=numpy.where(selected_boolean)[0] # indices    
    if len(selected)==0:
        print('There is no Runner_X.m call in the NetworkList.xlsx file: X')
    ## End
    return mfilenameAlg,mfilenameVer,AlgName,LocalFolder,LocalFolderFilesTemp,BSelecAlg,BSelecAlgVer,BSelecAlgName,InputFolderx,datar,selected

File: AI/code_01/library/specific/Matlab/FileNames_v01.m
function [fname,datar,selected,cverSubFile_a,cverSubFile_b,AISubFile_a,AISubFile_b,AlgName]=FileNames_v01(version)
fname=dir('*NetworkList*.xlsx');               % List of networks to be tested
datar=readmatrix(fname.name,'Range','A2:N50');
selected=find(datar(:,end)==version);

%%%
% Start (Manual entry required)
% Beam selection algorithm in Matlab. See the updated list in NetworkList_v01.xlsx-AlgorithmsList_v01 sheet
%cverSubFile_a='AI';   % Linear_Joint_Init_Iter_Rate_NoBCC
%cverSubFile_a='BI';   % Linear_Disjoint_DL_BCC
%cverSubFile_a='CI';   % Linear_Joint_Init_Iter_DL_BCC
cverSubFile_a='DI';   % Linear_Joint_Init_Iter_Rate_BCC
%cverSubFile_a='EI';   % Semilinear_Joint_Init_Iter_Rate_BCC
%cverSubFile_a='FI';   % Linear_Joint_Init_Iter_Sel_Rate_BCC
cverSubFile_b='v01';

% AI algorithm in Python. See the updated list in NetworkList_v01.xlsx-AlgorithmsList_v01 sheet
% AISubFile_a='AI'; % LSVM
% AISubFile_a='BI'; % GSVM
% AISubFile_a='CI'; % MLP
AISubFile_a='DI'; % RF
AISubFile_b='v01';
AlgNameB='ZF';                      % Digital filter type
% End (Manual entry required)
%%%

%%%
% Retrieve beam selection algorithm name from the file name
[~,FileName,~]=xlsread(fname.name,'AlgorithmsList_v01','C1:C30');
[~,AlgNames,~]=xlsread(fname.name,'AlgorithmsList_v01','D1:D30');
AlgNameAx=AlgNames(strcmp([cverSubFile_a '_' cverSubFile_b],FileName));
AlgNameA=AlgNameAx{1};
AlgName=[AlgNameA '_' AlgNameB];                    
%%%

% AI_v01.py	 LSVM
% BI_v01.py	 GSVM
% CI_v01.py	 MLP
% AI_v01.py	 RF